torch/utils/data/backward_compatibility.py

```diff
@@ -1,8 +1,20 @@
 import torch.utils.data.graph_settings
+import warnings
+import torch.distributed as dist
+import torch.distributed.distributed_c10d as c10d
 
 
 def worker_init_fn(worker_id):
+    warnings.warn("Usage of backward_compatibility.worker_init_fn is deprecated"
+                  " as DataLoader automatically calls it in every worker")
+
+
+def private_worker_init_fn(worker_id):
```

Comment:
IIUC, the only behavior is `torch.utils.data.graph_settings.apply_sharding()`. Can that be overridden? I think if users call `apply_sharding` again, it will just raise an exception.